includes/api/class-wc-rest-product-attribute-terms-controller.php

```diff
@@ -14,14 +14,98 @@
  * REST API Product Attribute Terms controller class.
  *
  * @package WooCommerce/API
- * @extends WC_REST_Product_Attribute_Terms_V2_Controller
+ * @extends WC_REST_Product_Attribute_Terms_V3_Controller
  */
-class WC_REST_Product_Attribute_Terms_Controller extends WC_REST_Product_Attribute_Terms_V2_Controller {
+class WC_REST_Product_Attribute_Terms_Controller extends WC_REST_Product_Attribute_Terms_V3_Controller {
 
 	/**
 	 * Endpoint namespace.
 	 *
 	 * @var string
 	 */
-	protected $namespace = 'wc/v3';
+	protected $namespace = 'wc/v4';
+
+	/**
+	 * Prepare a single product attribute term output for response.
+	 *
+	 * @param WP_Term $item Term object.
+	 * @param WP_REST_Request $request
+	 * @return WP_REST_Response $response
+	 */
+	public function prepare_item_for_response( $item, $request ) {
+		// Get term order.
+		$menu_order = get_woocommerce_term_meta( $item->term_id, 'order_' . $this->taxonomy );
+
+		// Get the attribute slug.
+		$attribute_id = absint( $request->get_param( 'attribute_id' ) );
+		$attribute    = wc_get_attribute( $attribute_id );
+
+		$data = array(
+			'id'          => (int) $item->term_id,
+			'name'        => $item->name,
+			'slug'        => $item->slug,
+			'description' => $item->description,
+			'menu_order'  => (int) $menu_order,
+			'count'       => (int) $item->count,
+			'attr_name'   => $attribute->name,
+			'attr_slug'   => $attribute->slug,
+		);
+
+		$context = ! empty( $request['context'] ) ? $request['context'] : 'view';
+		$data    = $this->add_additional_fields_to_object( $data, $request );
+		$data    = $this->filter_response_by_context( $data, $context );
+
+		$response = rest_ensure_response( $data );
+
+		$response->add_links( $this->prepare_links( $item, $request ) );
+
+		/**
+		 * Filter a term item returned from the API.
+		 *
+		 * Allows modification of the term data right before it is returned.
+		 *
+		 * @param WP_REST_Response  $response  The response object.
+		 * @param object            $item      The original term object.
+		 * @param WP_REST_Request   $request   Request used to generate the response.
+		 */
+		return apply_filters( "woocommerce_rest_prepare_{$this->taxonomy}", $response, $item, $request );
+	}
+
+	/**
+	 * Get the Attribute Term's schema, conforming to JSON Schema.
+	 *
+	 * @return array
+	 */
+	public function get_item_schema() {
+		$raw_schema = parent::get_item_schema();
+		$schema     = array(
+			'$schema'    => 'http://json-schema.org/draft-04/schema#',
+			'title'      => 'product_block_category',
```

Comment:
We can't use this `title` anymore, need to be `product_attribute_term`.

Also we usually don't use short names in our schema, I prefer if we change `attr_name` and `attr_slug` to:

```php
		$schema['properties']['attribute']   = array(
			'description' => __( 'Attribute.', 'woocommerce' ),
			'type'        => 'string',
			'context'     => array( 'view', 'edit' ),
			'readonly'    => true,
			'items'       => array(
				'type'       => 'object',
				'properties' => array(
					'id' => array(
						'description' => __( 'Attribute ID.', 'woocommerce' ),
						'type'        => 'integer',
						'context'     => array( 'view', 'edit' ),
						'readonly'    => true,
					),
					'name' => array(
						'description' => __( 'Attribute name.', 'woocommerce' ),
						'type'        => 'string',
						'context'     => array( 'view', 'edit' ),
						'readonly'    => true,
					),
					'slug' => array(
						'description' => __( 'Attribute slug.', 'woocommerce' ),
						'type'        => 'string',
						'context'     => array( 'view', 'edit' ),
						'readonly'    => true,
					),
				),
			),
		);
```

So we can group all information and also include ID to help fetch more data about the attribute.